OOInteraction/src/commands/CSceneHandlerItemTest.cpp

```diff
@@ -53,76 +56,133 @@ bool CSceneHandlerItemTest::canInterpret(Visualization::Item*, Visualization::It
-	QString oldVersionProject = "MyTest"+oldCommitId;
+	// load name into tree
+	std::unique_ptr<const FilePersistence::Commit> oldCommit{repository.getCommit(oldCommitId)};
+	std::unique_ptr<const FilePersistence::Commit> newCommit{repository.getCommit(newCommitId)};
 
-	FilePersistence::SimpleTextFileStore store;
-	store.setBaseFolder(projectsDir);
+	auto oldFileStore = new FilePersistence::SimpleTextFileStore{
+				[this, &oldCommit](QString filename, const char*& data, int& size)
+				{ return oldCommit->getFileContent(filename, data, size, false); }
+			};
+	auto newFileStore = new FilePersistence::SimpleTextFileStore{
+			[this, &newCommit](QString filename, const char*& data, int& size)
+			{ return newCommit->getFileContent(filename, data, size, false); }
+		};
 
 	// load newer version
 	Model::TreeManager* newVersionManager = new Model::TreeManager{};
-	newVersionManager->load(&store, newVersionProject, false);
+	newVersionManager->load(newFileStore, projectName, false);
 
 	// load older version
 	Model::TreeManager* oldVersionManager = new Model::TreeManager{};
-	oldVersionManager->load(&store, oldVersionProject, false);
+	oldVersionManager->load(oldFileStore, projectName, false);
 
 	Model::Reference::resolvePending();
 
-	// get GitRepository
-	QString path{projectsDir + newVersionProject};
-	if (!FilePersistence::GitRepository::repositoryExists(path))
-	{
-		qDebug() << "no repository found at " + path;
-		return new Interaction::CommandResult{};
-	}
-
-	FilePersistence::GitRepository repository{path};
-
 	FilePersistence::Diff diff = repository.diff(oldCommitId, newCommitId);
+
 	auto changes = diff.changes();
 
 	QSet<Model::NodeIdType> changedNodesToVisualize;
-
-	auto typeMatcher = Model::SymbolMatcher::guessMatcher("Class");
+	QSet<Model::NodeIdType> movedNodes;
+	QSet<Model::NodeIdType> deletedNodes;
+	QSet<Model::NodeIdType> insertedNodes;
+	QSet<Model::NodeIdType> modifiedNodes;
 
 	for (auto change : changes.values())
 	{
 		if (change->isFake() || change->onlyStructureChange()) continue;
 
 		auto id = change->nodeId();
 
-		// search one instance of the changed node
-		if (auto node = const_cast<Model::Node*>(Model::AllTreeManagers::instance().nodeForId(id)))
+		switch (change->type())
+		{
+			case FilePersistence::ChangeType::Deletion:
+				qDebug() << "Deletion";
+				deletedNodes.insert(id);
+				break;
+			case FilePersistence::ChangeType::Insertion:
+				qDebug() << "Insertion";
+				insertedNodes.insert(id);
+				break;
+			case FilePersistence::ChangeType::Move:
+				qDebug() << "Move";
+				movedNodes.insert(id);
+				break;
+			case FilePersistence::ChangeType::Stationary:
+				qDebug() << "Stationary";
+				modifiedNodes.insert(id);
+				break;
+			case FilePersistence::ChangeType::Unclassified:
+				qDebug() << "Unclassified";
+				break;
+		}
+
+
+		// search one instance of the changed node NEW VERSION MANAGER
+		if (auto node = const_cast<Model::Node*>(newVersionManager->nodeIdMap().node(id)))
```

Comment:
This `if` is very similar to the one below. You could easily create a method (or a lambda) that unifies their behavior.

In general you should always avoid code duplication. In this particular case, since this is in the test command, you can leave it as it is, if you don't plan on using it any more, but otherwise please reduce duplication.